src/hooks/v2/contractReader/UserHasPermission.ts

```diff
@@ -0,0 +1,34 @@
+import { NetworkContext } from 'contexts/networkContext'
+import { useContext } from 'react'
+import { V2OperatorPermission } from 'models/v2/permissions'
+import { V2ProjectContext } from 'contexts/v2/projectContext'
+
+import useProjectOwner from './ProjectOwner'
+import { useHasPermissions } from './HasPermissions'
+
+export function useUserHasPermission(
+  permission: V2OperatorPermission | V2OperatorPermission[],
+) {
+  const { userAddress } = useContext(NetworkContext)
+  const { projectId, isPreviewMode } = useContext(V2ProjectContext)
+
+  const { data: owner } = useProjectOwner(projectId)
+
+  const hasOperatorPermission = useHasPermissions({
+    operator: userAddress,
+    account: owner,
+    domain: projectId,
+    permissions: Array.isArray(permission) ? permission : [permission],
+  })
```

Comment:
This file was renamed from `HasPermission` to `UserHasPermission`

This is the only meaningful code change. We use the new `useHasPermissions` hook internally.

Reply:
seems like we have a pattern of using V1/V2 in filenames that have both a V1 and V2 version. maybe these hook names should include V1/V2? would feel more explicit than identifying them just by their import paths